apm-agent-plugins/apm-aws-sdk/apm-aws-sdk-common/src/main/java/co/elastic/apm/agent/awssdk/common/AbstractMessageIteratorWrapper.java

```diff
@@ -93,12 +99,20 @@ public void endMessageProcessingSpan() {
 
     @Override
     public Message next() {
-        Transaction<?> currentTransaction = endCurrentTransaction();
-        Message sqsMessage = delegate.next();
-        if (currentTransaction == null) {
-            sqsInstrumentationHelper.startTransactionOnMessage(sqsMessage, queueName, textHeaderGetter);
+        try {
```

Comment:
For reviewer: this handles the case when hasNext()/next() is not used for the app loop, and just next() is looped and the exception caught

Reply:
I'm not sure to remember this in the future, see my suggested comment below.